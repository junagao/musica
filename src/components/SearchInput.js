import React from 'react';
import PropTypes from 'prop-types';

import './SearchInput.scss';

const SearchInput = ({ onSearchTermChange }) => (
  <label className="search-field" htmlFor="search">
    <input
      type="search"
      onChange={onSearchTermChange}
      name="search"
      placeholder="Search"
      className="search-input"
    />
  </label>
);

SearchInput.propTypes = {
  onSearchTermChange: PropTypes.func.isRequired,
};

export default SearchInput;
