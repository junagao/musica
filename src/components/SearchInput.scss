.search-input {
  color: #553c9a;
  border: 1px solid #553c9a;
  border-radius: 0.5rem;
  font-size: 1.5rem;
  margin-right: 0.5rem;
  padding: 0.75rem;
  width: 17rem;
  -webkit-appearance: none;
  &:focus {
    color: #553c9a;
    background-color: #faf5ff;
    border: 1px solid #553c9a;
    box-shadow: 0 1px 1px 0 #e9d8fd, 0 1px 1px 0 #d6bcfa;
    outline: none;
  }
  &::-webkit-input-placeholder {
    color: #d0bfe4;
  }
}
