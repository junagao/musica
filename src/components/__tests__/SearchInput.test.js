import React from 'react';
import { shallow } from 'enzyme';

import SearchInput from 'components/SearchInput';

describe('SearchInput', () => {
  let wrapper;
  const onSearchTermChangeMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <SearchInput onSearchTermChange={onSearchTermChangeMock} />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one label element with className of search-field and htmlFor property of search', () => {
    expect(wrapper.find('label').length).toEqual(1);
    expect(wrapper.find('label').hasClass('search-field')).toBeTruthy();
    expect(wrapper.find('label').prop('htmlFor')).toEqual('search');
  });

  it('should render one input element of type `search`, className `search-input`, name `search`, and placeholder property `Search`', () => {
    expect(wrapper.find('input').length).toEqual(1);
    expect(wrapper.find('input').hasClass('search-input')).toBeTruthy();
    expect(wrapper.find('input').prop('type')).toEqual('search');
    expect(wrapper.find('input').prop('name')).toEqual('search');
    expect(wrapper.find('input').prop('placeholder')).toEqual('Search');
  });

  it('should call onSearchTermChange prop passing the searchTerm', () => {
    const searchTerm = 'the national';
    wrapper.find('input').simulate('change', searchTerm);
    expect(onSearchTermChangeMock).toBeCalledWith('the national');
  });
});
