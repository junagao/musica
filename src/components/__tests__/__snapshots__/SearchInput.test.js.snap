// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should render correctly given the required props 1`] = `
<label
  className="search-field"
  htmlFor="search"
>
  <input
    className="search-input"
    name="search"
    onChange={[MockFunction]}
    placeholder="Search"
    type="search"
  />
</label>
`;
